### PR TITLE
create sidebar bottom div with latest update date

### LINKED ISSUE #257: add "last updated: date / time" to sidebar
at the bottom of the sidebar, add a field that displays "data last updated: day month year" date for users

the date should reflect the last time circle CI successfully ran (ie the last time the pipeline updated from the city's dataset)

### PR DESCRIPTION
**Pull request by mrtmrt233**
resolves #257 

# Motivation and context
seem like a quick enough piece to knock out

# What I did
- add a div to bottom of sidebar in index.html, and corresponding css in sidebar.css
- add function fillLastUpdate() to Sidebar.js, called in Sidebar.showDefault() to update info
- within fillLastUpdate(): google cloud storage api was called, and returned xml was parsed to fill in the last updated date

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -233,6 +233,8 @@ <h5>About | Share Feedback | Report a Missing Tree  | <a href = "https://github.
         <h1>Error</h1>
         <div>The tree you requested was not found.</div>
       </div>
+      <div id="sidebar-last-updated">Last updated :
```
Comment (emillipede) [inline public/index.html:236]: please add data ("Data last updated: ## month year)

**Comment by emillipede** [comment]
thank you @mrtmrt233 !

instead of adding a div, please move "last updated" text to be part of the information panel, ie text within `populateTreePanel` (and vacant version)

it'll be more intelligible (on mobile and desktop) if the alignment and styling of this text is consistent with the existing sidebar content, and this change will avoid confusing overlaps like this one:

Screen Shot 2020-04-22 at 9 04 40 AM

**Comment by mrtmrt233** [comment]
Adjusted as required. 
- the last-update div is now part of sidebar-content
- css adjusted
- api fetch now only run on app initialization
- date string format adjusted

File: public/js/Sidebar.js [hunk 2]
```diff
@@ -306,6 +309,48 @@ var initialY = null;
...
+      for (item of items) {
+        if (item.children[0].innerHTML === 'data/map.json') {
+          //convert format
+          this.lastUpdate = new Date(item.children[3].innerHTML)
+          const options = {year: 'numeric', month: 'short', day: 'numeric' }
+          let elements = document.getElementsByClassName('sidebar-last-updated')
+          //update DOM
+          for (i of elements) {
+            //date format: dd MON, YYYY
+            i.innerHTML = 'Data last Updated: ' + this.lastUpdate.toLocaleDateString('en-GB', options)
```
Comment (emillipede) [inline public/js/Sidebar.js:342]: please remove the uppercase U

File: public/index.html [hunk 3]
```diff
@@ -228,6 +232,8 @@ <h1>Explore Santa Monica's Urban Forest</h1>
         Public Tree Map will soon share information about the environmental benefits each tree provides — how many gallons of stormwater it intercepts, how many kilowatt-hours of energy it conserves, how many pounds of air pollutants it captures, and how many tons of carbon dioxide it removes from the atmosphere.</div>
         <h5>About | Share Feedback | Report a Missing Tree  | <a href = "https://github.com/Public-Tree-Map/public-tree-map-data-pipeline#data-sources">Data Sources</a></h5>
         <button id="sidebar-default-explore" class = "btn btn--mobile">Explore the Map</button>
+        <div class="sidebar-last-updated">Last updated : 
+        </div>
```
Comment (emillipede) [inline public/index.html:236]: please remove the section from this version of the sidebar (the tree/vacant panels)

**Comment by emillipede** [comment]
thank you again @mrtmrt233 ! just so I can merge it, I made those two small changes.

**Pull request merged**